Modules/GLO/src/ITSTPCMatchingTask.cxx

```diff
@@ -99,72 +106,7 @@ void ITSTPCMatchingTask::initialize(o2::framework::InitContext& /*ctx*/)
-    getObjectsManager()->startPublishing(mMatchITSTPCQC.getFractionITSTPCmatchPhiVsPt(matchType(i)));
-
-    // Eta
-    getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoEtaNum(matchType(i)));
-    getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoEtaDen(matchType(i)));
-    getObjectsManager()->startPublishing(mMatchITSTPCQC.getFractionITSTPCmatchEta(matchType(i)));
-
-    getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoEtaVsPtNum(matchType(i)));
-    getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoEtaVsPtDen(matchType(i)));
-    getObjectsManager()->startPublishing(mMatchITSTPCQC.getFractionITSTPCmatchEtaVsPt(matchType(i)));
-
-    // 1/Pt
-    getObjectsManager()->startPublishing(mMatchITSTPCQC.getHisto1OverPtNum(matchType(i)));
-    getObjectsManager()->startPublishing(mMatchITSTPCQC.getHisto1OverPtDen(matchType(i)));
-    getObjectsManager()->startPublishing(mMatchITSTPCQC.getFractionITSTPCmatch1OverPt(matchType(i)));
-
-    if (mMatchITSTPCQC.getUseMC()) {
-      // Pt
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoPtPhysPrimNum(matchType(i)));
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoPtPhysPrimDen(matchType(i)));
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getFractionITSTPCmatchPhysPrim(matchType(i)));
-
-      // Phi
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoPhiPhysPrimNum(matchType(i)));
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoPhiPhysPrimDen(matchType(i)));
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getFractionITSTPCmatchPhiPhysPrim(matchType(i)));
-
-      // Eta
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoEtaPhysPrimNum(matchType(i)));
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoEtaPhysPrimDen(matchType(i)));
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getFractionITSTPCmatchEtaPhysPrim(matchType(i)));
-
-      // 1/Pt
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getHisto1OverPtPhysPrimNum(matchType(i)));
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getHisto1OverPtPhysPrimDen(matchType(i)));
-      getObjectsManager()->startPublishing(mMatchITSTPCQC.getFractionITSTPCmatchPhysPrim1OverPt(matchType(i)));
-    }
-  }
-  // Residuals
-  getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoResidualPt());
-  getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoResidualPhi());
-  getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoResidualEta());
-
-  getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoChi2Matching());
-  getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoChi2Refit());
-  getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoTimeResVsPt());
-
-  getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoDCAr());
+  mMatchITSTPCQC.publishHistograms(getObjectsManager());
```

Comment:
Is `publishHistograms` used somewhere else ? or just in QC ?

Reply:
I think just in QC, it is not the only way to access the histogram and publish them, however like this one does not need to have 2 PRs in order to publish all the histograms and one in the O2 is enough.

Reply:
ok understood